src/abstractdataframe/abstractdataframe.jl

```diff
@@ -197,14 +201,7 @@ eltypes(df)
 ```
 
 """
-function eltypes(df::AbstractDataFrame)
-    ncols = size(df, 2)
-    res = Array(Type, ncols)
-    for j in 1:ncols
-        res[j] = eltype(df[j])
-    end
-    return res
-end
+eltypes(df::AbstractDataFrame) = map!(eltype, Vector{Type}(size(df,2)), columns(df))
```

Comment:
I have tried several variants, including `map()` and comprehensions. Unfortunately, `eltype()` returns `Any`, so it results in "reds" for `@code_warntype eltypes(..)`. The proposed version gives the shortest "no-reds" code.